src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -428,31 +453,25 @@ protected override Task AppendProjectStateDiagnosticsAsync(Project project, Docu
 
             private async Task<CompilationWithAnalyzers> GetCompilationWithAnalyzersAsync(Project project, bool concurrentAnalysis, bool reportSuppressedDiagnostics, CancellationToken cancellationToken)
             {
-                CompilationWithAnalyzers compilationWithAnalyzers;
-                lock (_compilationWithAnalyzersMap)
-                {
-                    if (_compilationWithAnalyzersMap.TryGetValue(project.Id, out compilationWithAnalyzers))
-                    {
-                        return compilationWithAnalyzers;
-                    }
-                }
-
-                if (project.SupportsCompilation)
+                if (_compilationWithAnalyzers == null && project.SupportsCompilation)
```

Comment:
@basoundr we really need a rolling perf test for diagnostics and especially in context of FixAll in project/solution. I regressed this in Update1 with v2 driver and this can easily happen again. Can we request perf team to focus on this next week?